### ISSUE TITLE
How to train with validation set (i.e. refit)

### ISSUE THREAD

**Issue by franklinwu19** [catboost/catboost#1147]
Hi community, I am working on catboost with time series data. I will set aside some data for evaluation purpose use `eval_set=(X, y)`. The question is that I will need these evaluation data, along with the training set to **re-train** the model for final predictions, something like `GridSearchCV` where you can set `refit=True`, it will use cross-validation to select best param and train on the whole dataset.
how can I implement that? Do I need to construct a new model class and transfer params, fit again with whole dataset?
Thanks for help!

**Comment by moritzwilksch** [comment]
Hey, did you find an elegant solution to this?

**Comment by benjamin-breton-loreal** [comment]
I am also interrested
